lib/sprockets/uri_utils.rb

```diff
@@ -137,15 +138,24 @@ def parse_file_digest_uri(uri)
     # Examples
     #
     #   build("/tmp/js/application.js")
-    #   # => "file-digest:/tmp/js/application.js"
+    #   # => "file-digest:///tmp/js/application.js"
     #
     # path - String file path
     #
     # Returns String URI.
     def build_file_digest_uri(path)
-      join_file_uri('file-digest'.freeze, nil, path, nil)
+      relative_path =  UnloadedAsset.new(path, self).relative_path
+      if relative_path == path
+        # path is absolute, use forward slash
```

Comment:
UnloadedAsset#relative_path may return an absolute path if the `uri `passed into it is outside of the `environment.root`. So when the original path matches the "relative path" it means that the "relative path" returned is in fact absolute. It's not ideal naming, i'm working on it.